### DISCUSSION TITLE
Issue connecting to DuckDB with JOOQ: "Binder Error: Referenced table 'system' not found"

### DISCUSSION THREAD

**Discussion by SreevidyaAravind1989** [jOOQ/jOOQ#16954] [Q&A] [answered]
Hello everyone,

I'm encountering an issue while trying to connect JOOQ to a DuckDB database. Here are the details:

Error message:
java.lang.RuntimeException: org.jooq.exception.DataAccessException: SQL [select system.main.duckdb_databases.database_name from system.main.duckdb_databases()]; java.sql.SQLException: Binder Error: Referenced table "system" not found!

I tried the following code.
        
String url = "jdbc:duckdb:memory";
        String user = "";
        String password = "";

        // Establish a connection to the DuckDB database
        try (Connection conn = DriverManager.getConnection(url, user, password)) {
            // Configure the code generation
            Configuration configuration = new Configuration()
                    .withJdbc(new Jdbc()
                            .withDriver("org.duckdb.DuckDBDriver")
                            .withUrl(url)
                            .withUser(user)
                            .withPassword(password))
                    .withGenerator(new Generator()
                            .withName("org.jooq.codegen.JavaGenerator")
                            .withDatabase(new Database()
                                    .withName("org.jooq.meta.duckdb.DuckDBDatabase")
                                    .withInputSchema("main"))
                            .withGenerate(new Generate()
                                    .withDaos(true)
                                    .withPojos(true))
                            .withTarget(new Target()
                                    .withPackageName("com.med.jooq")
                                    .withDirectory("target/generated-sources/jooq")));

            // Run the code generation
            GenerationTool.generate(configuration);

Has anyone encountered a similar issue with DuckDB and JOOQ? Any insights or recommendations on how to resolve this "Binder Error"?

Any help would be greatly appreciated. Thank you!

**Comment by lukaseder** [comment, marked as answer]
See https://github.com/jOOQ/jOOQ/issues/16869
